crates/astria-core/src/primitive/v1/asset/denom.rs

```diff
@@ -509,6 +516,72 @@ impl FromStr for IbcPrefixed {
     }
 }
 
+#[cfg(feature = "serde")]
+mod serde_impl {
+    use serde::{
+        Deserialize,
+        Deserializer,
+        Serialize,
+        Serializer,
+    };
+
+    macro_rules! impl_serde {
+        ($($type:ty),*$(,)?) => {
+            $(
+                impl<'de> Deserialize<'de> for $type {
+                    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+                    where
+                        D: Deserializer<'de>,
+                    {
+                        use serde::de::Error as _;
+                        let s = <&str>::deserialize(deserializer)?;
+                        s.parse().map_err(D::Error::custom)
+                    }
+                }
+
+                impl Serialize for $type {
+                    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+                    where
+                        S: Serializer,
+                    {
+                        serializer.collect_str(self)
+                    }
+                }
+            )*
+        }
+    }
+
+    impl_serde!(super::Denom, super::TracePrefixed, super::IbcPrefixed);
```

Comment:
are these supposed to be in this PR? seems somewhat unrelated

Reply:
Yeah, this is needed because `GenesisState` relies entirely on derived serde impls contains straight `asset::Denom` (either directly as in `allowed_fee_assets: Vec<asset::Denom>` or through the sequencer-defined `Account`).